### DISCUSSION TITLE
Is it supported to give a delayed transition an id?

### DISCUSSION THREAD

**Discussion by tklepzig** [statelyai/xstate#1502] [General] [answered]
In order to cancel a delayed transition, I add an id to it and it seems to work fine, see my machine in the visualizer: https://xstate.js.org/viz/?gist=3be59851512f44a340a4cd493a544308

I used it in the visualizer only (for [another issue](https://github.com/davidkpiano/xstate/issues/1501)) but afterwards, I implemented the machine in typescript and the typescript compiler complains about the `id` property inside my delay object `5000: {target: "disabled", id: "delay"}`. Now my question: Is it even supported officially?

**Comment by Andarist** [comment, marked as answer]
No, this is not currently supported. This is how delayed transitions are actually "translated" into "regular" transitions: https://github.com/davidkpiano/xstate/blob/37d5ed8606833f900a87f2a7de8f3f5059cf4deb/packages/core/src/StateNode.ts#L581-L586

**Reply by tklepzig** [reply]
Alright, thank you!
